src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -43,12 +42,12 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
             // Channels
             services.RegisterOutputChannel();
 
-            // Internal logging            
-            services.AddLogging(logging =>
-            {
-                logging.Services.AddSingleton<ILoggerProvider, GrpcFunctionsHostLoggerProvider>();
-                logging.Services.AddSingleton<IWorkerDiagnostics, GrpcWorkerDiagnostics>();
-            });
+            // Internal logging
+            services.AddSingleton<GrpcFunctionsHostLogWriter>();
+            services.AddSingleton<IUserLogWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
+            services.AddSingleton<ISystemLogWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
+            services.AddSingleton<IUserMetricWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
+            services.AddSingleton<IWorkerDiagnostics, GrpcWorkerDiagnostics>();
```

Comment:
Unrelated to changes from this PR: 
Are we supposed to call ` _outputChannel.TryWrite(message);` [here?](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Grpc/Http/GrpcWorkerDiagnostics.cs#L70)

Reply:
Certainly seems like it... I can follow up on that with a separate PR since it's unrelated.